### PR TITLE
test(mockwebserver): verify ANY method expectations

### PR DESCRIPTION
**Pull request by manusa**
## Description

Part of #5632

test(mockwebserver): verify ANY method expectations

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/main/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: junit/mockwebserver/src/test/groovy/io/fabric8/mockwebserver/DefaultMockServerAnyMethodTest.groovy [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    client.close()
+  }
+
+  def cleanupSpec() {
+    vertx.close()
+  }
+
+  def "when setting an expectation for ANY should respond to POST"() {
+    given: "An expectation with once"
+      server.expect().any().withPath("/api/v1/resource").andReturn(200, "OK").always()
```
Comment (rohanKanojia) [inline junit/mockwebserver/src/test/groovy/io/fabric8/mockwebserver/DefaultMockServerAnyMethodTest.groovy:51]: Umm, I might be wrong here but this statement doesn't seem to match what the description says. Should it be using `once()` instead?

File: junit/mockwebserver/src/test/groovy/io/fabric8/mockwebserver/DefaultMockServerAnyMethodTest.groovy [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        }
+      }
+
+    then: "Expect the result to be completed in the specified time"
+      async.await(10)
+  }
+
+  def "when setting an expectation for ANY should respond to PUT"() {
+    given: "An expectation with once"
+      server.expect().any().withPath("/api/v1/resource").andReturn(200, "OK").always()
```
Comment (rohanKanojia) [inline junit/mockwebserver/src/test/groovy/io/fabric8/mockwebserver/DefaultMockServerAnyMethodTest.groovy:72]: https://github.com/fabric8io/kubernetes-client/pull/5670/files#r1435096191

File: junit/mockwebserver/src/test/groovy/io/fabric8/mockwebserver/DefaultMockServerAnyMethodTest.groovy [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    and: "A POST request"
```
Comment (rohanKanojia) [inline junit/mockwebserver/src/test/groovy/io/fabric8/mockwebserver/DefaultMockServerAnyMethodTest.groovy:73]:
```suggestion
    and: "A PUT request"
```

**Review by oscerd — APPROVED** [review]

File: junit/mockwebserver/src/test/groovy/io/fabric8/mockwebserver/DefaultMockServerAnyMethodTest.groovy (hunk 1 above)
Comment (manusa) [reply junit/mockwebserver/src/test/groovy/io/fabric8/mockwebserver/DefaultMockServerAnyMethodTest.groovy:51]: will update after merging https://github.com/fabric8io/kubernetes-client/pull/5671

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**